app/features/user/auth.ts

```diff
@@ -43,14 +51,67 @@ type AuthState = {
     }
 );
 
+const userRepository = new UserRepository(boardwalkDb);
+
+const useUpsertBoardwalkUser = (openSecretUserData?: OpenSecretUser | null) => {
+  const [error, setError] = useState<unknown | null>(null);
+  const [boardwalkUserData, setBoardwalkUserData] = useState<{
+    loading: boolean;
+    user: BoardwalDbkUser | null;
+  }>({ loading: !!openSecretUserData, user: null });
+
+  useEffect(() => {
+    const upsertBoardwalkUser = async () => {
+      if (!openSecretUserData) {
+        return;
+      }
+
+      try {
+        const boardwalkUser = await userRepository.upsert(openSecretUserData);
+        setBoardwalkUserData({ loading: false, user: boardwalkUser });
+      } catch (e) {
+        setError(e);
+      }
+    };
+    upsertBoardwalkUser();
+  }, [openSecretUserData]);
+
+  if (error) {
+    throw error;
```

Comment:
if upserting data to postgres fail this will throw and then error boundary will show error message. this should happen only if our db is unreachable for some reason